### PR TITLE
Add document outlining the differences between probe-cli and probe-le…

### PR DESCRIPTION
**Pull request by hellais**
…gacy

### CODE REVIEW

File: docs/OONIProbeLegacyCompatibility.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Supported: ✅
+probe-cli equivalent: `--config`
+
+`-d, --datadir` Specify a path to the ooniprobe data directory.
+Supported: ✅
+probe-cli equivalent: set the `OONI_HOME` environment variable
+
+`-a, --annotations key:value[,key2:value2]` Annotate the report with a key:value[, key:value] format.
+Supported: ✅
+Priority: high
```
Comment (bassosimone) [inline docs/OONIProbeLegacyCompatibility.md]: Confusing: why add priority if supported? Also: you sure it's supported?

Comment (bassosimone) [reply docs/OONIProbeLegacyCompatibility.md]: Suggestion: `-a key:value -a key2:value2` is easier to handle than what we had previously. (MK uses `-A` for that precisely because the probe-legacy behaviour is not implemented.)

Comment (bassosimone) [reply docs/OONIProbeLegacyCompatibility.md]: (xref: https://github.com/ooni/probe-engine/issues/132)

File: docs/OONIProbeLegacyCompatibility.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+itself and not expose this setting.
+
+# Features
+
+* Run a test deck
+Supported: ✅
+probe-cli equivalent: we now call a test deck a test group and we have them
+coded into the logic of the client
+
+* Run an individual netttest
```
Comment (bassosimone) [inline docs/OONIProbeLegacyCompatibility.md]: remove the third `t`, please

File: docs/OONIProbeLegacyCompatibility.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+* Test an individual URL with web_connectivity
+Supported: ❌
+Priority: high
+
+* Run tests automatically like `ooniprobe-agent`
+Supported: ❌
+Priority: high
+
+* Write custom tests like OONI test templates
```
Comment (bassosimone) [inline docs/OONIProbeLegacyCompatibility.md]: I don't understand what you mean here

File: docs/OONIProbeLegacyCompatibility.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+Supported: ❌
+Priority: medium
+
+* Packet captures
```
Comment (bassosimone) [inline docs/OONIProbeLegacyCompatibility.md]: I understood from above it was wontfix?

File: docs/OONIProbeLegacyCompatibility.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+probe-cli equivalent: set the `OONI_HOME` environment variable
+
+`-a, --annotations key:value[,key2:value2]` Annotate the report with a key:value[, key:value] format.
+Supported: ✅
+Priority: high
+
+`-P, --preferred-backend onion|https|cloudfront` Set the preferred backend to use when submitting results and/or communicating with test helpers. Can be either onion, https or cloudfront
+Supported: ❌
+Priority: wontfix, we don't support any other backend beyond https, yet we will
+and yet we would prefer to have the logic of reporting be managed by the probe
```
Comment (anadahz) [inline docs/OONIProbeLegacyCompatibility.md]: This seems like a syntax error: `yet we will and yet we would prefer`

File: docs/OONIProbeLegacyCompatibility.md (hunk 1 above)
Comment (hellais) [reply docs/OONIProbeLegacyCompatibility.md]: Ah yes you are right, this is not supported.

File: docs/OONIProbeLegacyCompatibility.md (hunk 3 above)
Comment (hellais) [reply docs/OONIProbeLegacyCompatibility.md]: I mean that we don't have a public API for writing tests like we had for OONI Probe 2.0.0, see: https://github.com/ooni/probe-legacy/blob/master/docs/source/writing_tests.rst

File: docs/OONIProbeLegacyCompatibility.md (hunk 5 above)
Comment (hellais) [reply docs/OONIProbeLegacyCompatibility.md]: Yeah there was an extra yet probably. I removed it.

File: docs/OONIProbeLegacyCompatibility.md (hunk 4 above)
Comment (hellais) [reply docs/OONIProbeLegacyCompatibility.md]: Maybe, I would not exclude it a-priori though. What do you think?

File: docs/OONIProbeLegacyCompatibility.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+Supported: ❌
+Priority: medium
+
+* Measurement quota to limit the disk usage
+Supported: ❌
+Priority: high
+
+* Failover strategies for uploading measurements (use https then onion then cloudfront)
+Supported: ❌
+Priority: high
```
Comment (anadahz) [inline docs/OONIProbeLegacyCompatibility.md]: Doesn't this contradict with?: https://github.com/ooni/probe-cli/pull/82/files#diff-a7b5bc713d5d56af7ba54f408fd8a4d7R67-R71
>wontfix, we don't support any other backend beyond https

**Comment by anadahz** [comment]
Some options that are not included in the outline:

* -v, --verbose           Show more verbose information
* -Q, --queue=           AMQP Queue URL,  amqp://user:pass@host:port/vhost/queue
* --version                 Display the ooniprobe version and exit.

Comments:
* Are the configuration options (in config file) in scope of this outline?
* Can you create/run your own custom test-list (not deck) in probe-cli?
* It will be very useful to see the available tests or tests that are not going to be implemented in probe-cli

File: docs/OONIProbeLegacyCompatibility.md (hunk 6 above)
Comment (hellais) [reply docs/OONIProbeLegacyCompatibility.md]: We will support failover strategies, but we will not expose settings for choosing this as the logic is to be handled by the probe itself, see:
> we would prefer to have the logic of reporting be managed by the probe itself and not expose this setting.

**Comment by hellais** [comment]
> Some options that are not included in the outline:

Thanks added these in: https://github.com/ooni/probe-cli/pull/82/commits/e4ff1fe89a7a5a29ee48d80dab9b649f692309d3

> Are the configuration options (in config file) in scope of this outline?

Yes config options are in scope. I believe the missing relevant ones should all be documented inside of the features section.

If you think some are missing, let me know.

I did not document stuff which is not relevant like the measurement and reporting concurrency, but if you think it's important to document those too, feel free to open a PR with that too. They are likely `wontfix`, unless there is a strong argument in their favour.

> Can you create/run your own custom test-list (not deck) in probe-cli?

No that is not supported. See: https://github.com/ooni/probe-cli/issues/37 and in this file:
> * Test an individual URL with web_connectivity

I added an extra item just for test lists from files.

> It will be very useful to see the available tests or tests that are not going to be implemented in probe-cli

You can find all the supported tests in here: https://github.com/ooni/probe-engine/blob/master/experiment.go#L269.

Basically it's all the tests inside of the mobile app + psiphon. You can find the list of tests we have plans for in here: https://github.com/ooni/probe-engine/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+label%3A%22new+experiment%22+ or https://github.com/ooni/spec/issues?q=is%3Aissue+is%3Aopen+label%3A%22new+test%22.

If there is a test you think is especially important to implement give the issue a thumbs up! 👍

File: docs/OONIProbeLegacyCompatibility.md (hunk 4 above)
Comment (bassosimone) [reply docs/OONIProbeLegacyCompatibility.md]: Yeah, but then you should not flag it as wontfix above?

File: docs/OONIProbeLegacyCompatibility.md (hunk 3 above)
Comment (bassosimone) [reply docs/OONIProbeLegacyCompatibility.md]: Gotcha! Can you then maybe rephrase this to indicate we want to have a public API for writing tests?

File: docs/OONIProbeLegacyCompatibility.md (hunk 4 above)
Comment (hellais) [reply docs/OONIProbeLegacyCompatibility.md]: Ok I will put the above one as not wontfix then.

**Comment by hellais** [comment]
@bassosimone I have addressed you feedback and will now merge. Thanks for the review!
